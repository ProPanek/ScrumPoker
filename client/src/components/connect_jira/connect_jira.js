import React, {Component} from "react";
import {Grid, Card} from "@material-ui/core";
import styled from "styled-components";
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import {decorate, observable} from "mobx";
import routes from "routes"
import ConnectJiraLoggedIn from "./connect_jira_logged_in";
import ConnectJiraLoggedOut from "./connect_jira_logged_out";

const StyledGrid = styled(Grid)`
  &&{
  height: calc(100vh - 48px);
  overflow-y:auto;
  }
`;

const StyledCard = styled(Card)`
  &&{
  height: calc(100vh - 64px);
  }
`;

const FormWrapper = styled.div`
  @media only screen and (max-width: 768px) {
     width: 90%;
  }
  display: flex;
  flex-direction: column;
  width: 50%;
  margin: 0 auto;
  position: relative;
  top: calc(50vh - 350px);
`;


class ConnectJira extends Component {
  constructor(props) {
    super(props)
    this.board = false
    this.saveCredentials = true
    this.jiraLogin = ""
    this.jiraPassword = ""
    this.jiraSubdomain = ""
  }

  componentDidMount() {
    if (localStorage.getItem("jira-credentials") !== null && localStorage.getItem("jira-subdomains") !== null) {
      const data = JSON.parse(localStorage.getItem("jira-credentials"));
      const subdomains = JSON.parse(localStorage.getItem("jira-subdomains"));
      this.props.store.jiraStore.jiraLogin(subdomains[0], data.jiraLogin, data.jiraPassword)
    }
    window.onpopstate = this.onBackButtonEvent
  }

  onBackButtonEvent = (e) => {
    const {store: {userStore: {leaveRoom}}, history: {push}} = this.props
    leaveRoom()
    push(routes.root())
  };

  render() {
    const {store: {jiraStore: {jiraLoggedIn}}} = this.props
    return (
      <StyledGrid item xs={12}>
        <StyledCard>
          <FormWrapper>
            {jiraLoggedIn ? <ConnectJiraLoggedIn/> : <ConnectJiraLoggedOut/>}
          </FormWrapper>
        </StyledCard>
      </StyledGrid>
    );
  }
}

decorate(ConnectJira, {
  jiraLogin: observable,
  jiraPassword: observable,
  jiraSubdomain: observable,
  jiraSubdomainMore: observable,
  board: observable,
  subdomains: observable,
  saveCredentials: observable
});

export {ConnectJira}
export default inject("store")(withRouter(observer(ConnectJira)));
