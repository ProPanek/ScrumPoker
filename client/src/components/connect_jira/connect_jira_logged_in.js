import React, {Component} from "react";
import {Grid, Button} from "@material-ui/core";
import styled from "styled-components";
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import FormLabel from "@material-ui/core/FormLabel";
import Typography from "@material-ui/core/Typography";
import CircularProgress from "@material-ui/core/CircularProgress";
import AddCircle from "@material-ui/icons/AddCircle"
import {Logout} from "mdi-material-ui";
import {decorate, observable} from "mobx";
import Jira from "img/jira.png"
import Select from "react-select"
import routes from "routes"

const StyledButton = styled(Button)`
  &&{
  width:100%;
  }
`;

const StyledSelect = styled(Select)`
  width:100%;
  margin-top:40px;
  margin-bottom: 40px;
`;

const StyledCircularProgress = styled(CircularProgress)`
  &&{
  margin: 0 auto;
  }
`;

const StyledTypography = styled(Typography)`
  &&{
  text-align:center;
  position:relative;
  top:20%;
  }
`;

const StyledImg = styled.img`
  width: 100px;
  margin: 0 auto;
`;

const StyledGrid = styled(Grid)`
  margin-top: 40px;
`;

const StyledAddCircle = styled(AddCircle)`
  margin-left: 10px;
`;

const StyledLogout = styled(Logout)`
  margin-left: 10px;
`;

class ConnectJiraLoggedIn extends Component {
  constructor(props) {
    super(props)
    this.board = false
  }

  handleJiraLogout = () => {
    localStorage.removeItem("jira-credentials");
    localStorage.removeItem("jira-subdomains");
    this.props.store.jiraStore.logout()
  }

  goToRoom = () => {
    const {store: {jiraStore: {jiraLoggedIn, saveBoardId}, roomStore: {roomName, roomId}}, history: {push}} = this.props
    if (jiraLoggedIn) {
      saveBoardId(this.board)
    }
    push(routes.room(roomName, roomId))
  }

  handleChangeBoard = selectedElement => {
    this.board = selectedElement.value
    this.props.store.jiraStore.setBoardId(selectedElement.value)
  };

  render() {
    const {store: {jiraStore: {jiraBoardsFetching, jiraBoards}}} = this.props
    return (
      <>
        <StyledImg src={Jira} alt={"Jira logo"}/>
        <Typography variant="h3" align={"center"}>
          Jira Integration
        </Typography>
        <Typography>
          Select board you want to make an estimations for and go to your room
        </Typography>
        {jiraBoardsFetching &&
        <StyledCircularProgress/>}
        {jiraBoards.values.length > 0 &&
        !jiraBoardsFetching &&
        <>
          <FormLabel> Jira Board </FormLabel>
          <StyledSelect
            onChange={this.handleChangeBoard}
            options={jiraBoards.values.map((data) => {
              return {
                value: data.id,
                label: data.name
              }
            })}
          />
        </>}
        <StyledGrid container>
          <Grid item xs={5}>
            <StyledButton disabled={!this.board} color="primary" variant="contained" onClick={this.goToRoom}>
              Go to room
              <StyledAddCircle/>
            </StyledButton>
          </Grid>
          <Grid item xs={2}>
            <StyledTypography variant="h5"> OR </StyledTypography>
          </Grid>
          <Grid item xs={5}>
            <StyledButton color="primary" variant="contained" onClick={this.handleJiraLogout}>
              Logout from Jira
              <StyledLogout/>
            </StyledButton>
          </Grid>
        </StyledGrid>
      </>
    );
  }
}

decorate(ConnectJiraLoggedIn, {
  board: observable,
});

export {ConnectJiraLoggedIn}
export default inject("store")(withRouter(observer(ConnectJiraLoggedIn)));
