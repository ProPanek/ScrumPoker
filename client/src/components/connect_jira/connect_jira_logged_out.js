import React, {Component} from "react";
import {Grid, Button, TextField} from "@material-ui/core";
import styled from "styled-components";
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import Typography from "@material-ui/core/Typography";
import {ArrowUpBold, Login} from "mdi-material-ui";
import {decorate, observable} from "mobx";
import Jira from "img/jira.png"
import Checkbox from "@material-ui/core/Checkbox";
import FormControlLabel from "@material-ui/core/FormControlLabel/FormControlLabel";
import routes from "routes"

const StyledButton = styled(Button)`
  &&{
  width:100%;
  }
`;

const StyledTypography = styled(Typography)`
  &&{
  text-align:center;
  position:relative;
  top:20%;
  }
`;

const StyledGrid = styled(Grid)`
  margin-top: 40px;
`;

const StyledArrowUpBold = styled(ArrowUpBold)`
  margin-left: 10px;
`;

const StyledLogin = styled(Login)`
  margin-left: 10px;
`;

class ConnectJiraLoggedOut extends Component {
  constructor(props) {
    super(props)
    this.saveCredentials = true
    this.jiraLogin = ""
    this.jiraPassword = ""
    this.jiraSubdomain = ""
  }

  handleJiraLogin = () => {
    if (this.jiraLogin && this.jiraPassword && this.jiraSubdomain) {
      this.props.store.jiraStore.jiraLogin(this.jiraSubdomain, this.jiraLogin, this.jiraPassword)
      const subDomains = [this.jiraSubdomain]
      const data = {
        jiraLogin: this.jiraLogin,
        jiraPassword: this.jiraPassword
      }
      this.jiraLogin = this.jiraPassword = this.jiraSubdomain = ""
      if (this.saveCredentials) {
        localStorage.setItem("jira-subdomains", JSON.stringify(subDomains));
        localStorage.setItem("jira-credentials", JSON.stringify(data));
      }
    }
  }

  handleChangeJiraLogin = (e) => {
    this.jiraLogin = e.target.value
  }

  handleChangeJiraPassword = (e) => {
    this.jiraPassword = e.target.value
  }

  handleChangeJiraSubdomain = (e) => {
    this.jiraSubdomain = e.target.value
  }

  handleChangeJiraCredentials = (e) => {
    this.saveCredentials = e.target.checked
  }

  goToRoom = () => {
    const {store: {jiraStore: {jiraLoggedIn, saveBoardId}, roomStore: {roomName, roomId}}, history: {push}} = this.props
    if (jiraLoggedIn) {
      saveBoardId(this.board)
    }
    push(routes.room(roomName, roomId))
  }

  render() {
    return (
      <>
        <img style={{width: "100px", margin: "0 auto"}} src={Jira} alt={"Jira logo"}/>
        <Typography variant="h3" align={"center"}>
          Jira Integration
        </Typography>
        <Typography>
          Do you want to setup Jira integration? This will allow you to select issues and set ther story points
          directly from Planning Poker
        </Typography>
        <TextField
          id="subdomain-jira"
          label="Jira subdomain"
          placeholder="subdomain.atlassian.net"
          value={this.jiraSubdomain}
          onChange={this.handleChangeJiraSubdomain}
          type="text"
          margin="normal"
        />
        <TextField
          id="login-jira"
          label="Jira E-mail"
          value={this.jiraLogin}
          onChange={this.handleChangeJiraLogin}
          type="email"
          margin="normal"
        />
        <TextField
          id="password-jira"
          label="Jira api-key"
          value={this.jiraPassword}
          onChange={this.handleChangeJiraPassword}
          type="password"
          margin="normal"
        />
        <FormControlLabel
          control={
            <Checkbox
              id="save-credentials"
              checked={this.saveCredentials}
              onChange={this.handleChangeJiraCredentials}
              value="save"
              color="primary"
            />
          }
          label="Keep me logged in"
        />
        <StyledGrid container>
          <Grid item xs={5}>
            <StyledButton color="primary" variant="contained" onClick={this.handleJiraLogin}>
              Connect to Jira
              <StyledLogin/>
            </StyledButton>
          </Grid>
          <Grid item xs={2}>
            <StyledTypography variant="h5"> OR </StyledTypography>
          </Grid>
          <Grid item xs={5}>
            <StyledButton variant="contained" onClick={this.goToRoom}>
              Skip and go to room
              <StyledArrowUpBold/>
            </StyledButton>
          </Grid>
        </StyledGrid>
      </>
    );
  }
}

decorate(ConnectJiraLoggedOut, {
  jiraLogin: observable,
  jiraPassword: observable,
  jiraSubdomain: observable,
  jiraSubdomainMore: observable,
  subdomains: observable,
  saveCredentials: observable
});

export {ConnectJiraLoggedOut}
export default inject("store")(withRouter(observer(ConnectJiraLoggedOut)));
