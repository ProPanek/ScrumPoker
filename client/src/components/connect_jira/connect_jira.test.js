import React from "react";
import {shallow} from "enzyme";
import {ConnectJira} from "./connect_jira";

describe("CreateRoom", () => {
  it("renders without crashing", () => {
    const props = {
      store: {
        jiraStore:{}
      }
    }
    const component = shallow(<ConnectJira {...props}/>)
    expect(component).toMatchSnapshot()
  });
});