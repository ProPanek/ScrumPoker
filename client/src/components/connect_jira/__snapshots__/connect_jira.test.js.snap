// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRoom renders without crashing 1`] = `
<Styled(WithStyles(Grid))
  item={true}
  xs={12}
>
  <Styled(WithStyles(Card))>
    <styled.div>
      <inject-withRouter(ConnectJiraLoggedOut)-with-store />
    </styled.div>
  </Styled(WithStyles(Card))>
</Styled(WithStyles(Grid))>
`;
